src/Surfnet/StepupMiddleware/ApiBundle/Tests/Service/DeprovisionServiceTest.php

```diff
@@ -0,0 +1,144 @@
+use Surfnet\StepupMiddleware\ApiBundle\Identity\Entity\Identity;
+use Surfnet\StepupMiddleware\ApiBundle\Identity\Repository\IdentityRepository as ApiIdentityRepository;
+use Surfnet\StepupMiddleware\ApiBundle\Service\DeprovisionService;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Identity\Command\ForgetIdentityCommand;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Pipeline\Pipeline;
+
+class DeprovisionServiceTest extends TestCase
+{
+    /**
+     * @var DeprovisionService
+     */
+    private $deprovisionService;
+
+    /**
+     * @var m\LegacyMockInterface|m\MockInterface|Pipeline
+     */
+    private $pipeline;
+
+    /**
+     * @var m\LegacyMockInterface|m\MockInterface|ApiIdentityRepository
+     */
+    private $apiRepo;
+
+    /**
+     * @var m\LegacyMockInterface|m\MockInterface|IdentityRepository
+     */
+    private $eventRepo;
+
+    protected function setUp(): void
+    {
+        $this->pipeline = m::mock(Pipeline::class);
+        $this->apiRepo = m::mock(ApiIdentityRepository::class);
+        $this->eventRepo = m::mock(IdentityRepository::class);
+        $logger = m::mock(LoggerInterface::class);
+        $logger->shouldIgnoreMissing(); // Not going to verify every log message at this point
+        $this->deprovisionService = new DeprovisionService($this->pipeline, $this->eventRepo, $this->apiRepo, $logger);
+    }
+
+    protected function tearDown(): void
+    {
+        m::close();
+        parent::tearDown();
+    }
+
+    public function test_it_can_be_created()
+    {
+        $this->assertInstanceOf(DeprovisionService::class, $this->deprovisionService);
+    }
+
+    /**
+     * @group api-bundle
+     */
+    public function test_it_deals_with_non_exisiting_collab_user_id()
+    {
+        $this->apiRepo
+            ->shouldReceive('findOneByNameId')
+            ->with('urn:collab:person:example.com:maynard_keenan')
+            ->once()
+            ->andReturnNull();
+        $data = $this->deprovisionService->readUserData('urn:collab:person:example.com:maynard_keenan');
+
+        $this->assertTrue(is_array($data));
+        $this->assertEmpty($data);
+    }
+
+    /**
+     * @group api-bundle
+     */
+    public function test_it_can_return_data()
+    {
+        $identity = m::mock(Identity::class);
+        $identity->id = '0bf0b464-a5de-11ec-b909-0242ac120002';
+        $this->apiRepo
+            ->shouldReceive('findOneByNameId')
+            ->with('urn:collab:person:example.com:maynard_keenan')
+            ->once()
+            ->andReturn($identity);
+
+        $this->eventRepo->shouldReceive('obtainInformation')->andReturn(['status' => 'OK', 'data' => []]);
+
+        $data = $this->deprovisionService->readUserData('urn:collab:person:example.com:maynard_keenan');
+
+        $this->assertTrue(is_array($data));
+        $this->assertEquals($data['status'], 'OK');
+    }
+
+    public function test_deprovision_does_not_deprovision_when_user_is_not_found()
```

Comment:
The test does not perform any assertions, this leads to an incomplete test.
Is there a reason for this?